extractor/src/test/suite/scala/com/digitalasset/extractor/config/TemplateConfigSpec.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.extractor.config
+
+import com.digitalasset.extractor.config.Generators.genTemplateConfig
+import org.scalacheck.Gen
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+class TemplateConfigSpec extends FlatSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  behavior of TemplateConfig.getClass.getSimpleName
+
+  it should "be sortable" in forAll(Gen.nonEmptyListOf(genTemplateConfig)) { templateConfigs =>
+    whenever(templateConfigs.size > 1) {
+      val sorted = templateConfigs.sorted
+      sorted.sliding(2).forall {
+        case List(x, y) => x.compareTo(y) <= 0
+      } should ===(true)
+    }
+  }
+
+  it should "have deterministic sort" in forAll(Gen.nonEmptyListOf(genTemplateConfig)) { as =>
```

Comment:
Don't test this; instead, test _antisymmetry_ of the ordering: for all a, b: TemplateConfig, consider compare(a, b) and compare(b, a): the only acceptable patterns are

1. positive, negative,
2. negative, positive,
3. zero, zero

anything else fails. (Use the `signum` method on `Int` to make this test easier.)

This precisely implies the property you're trying to test without burying it so that it doesn't really get covered.

Alternatively, you can just empty out this file and use Scalaz:

```scala
  // checkLaws is defined in ImmArrayTest; just copy-paste for now
  checkLaws(ScalazProperties.order.laws[TemplateConfig])
```

Reply:
I don't have `scalaz.Order` instance for `TemplateConfig`

Reply:
Just use `scalaz.Order.fromScalaOrdering[TemplateConfig]` (let implicit resolution do its thing); that will use what you've defined.